File: Week 6/sentimental-readability/readability.py
# TODO
from cs50 import get_string

text = get_string("Text: ").strip()
num_words, num_letters, num_sentences = 0, 0, 0

# using index to use entered text to find wnumber of words, number of letters and sentences
for i in range(len(text)):
    if (i == 0 and text[i] != ' ') or (i != len(text) - 1 and text[i] == ' ' and text[i + 1] != ' '):
        num_words += 1
    if text[i].isalpha():
        num_letters += 1
    if text[i] == '.' or text[i] == '?' or text[i] == '!':
        num_sentences += 1

length = num_letters / num_words * 100
sentence = num_sentences / num_words * 100

# equation to find grade level
index = round(0.0588 * length - 0.296 * sentence - 15.8)
if index < 1:
    print("Before Grade 1")
elif index >= 16:
    print("Grade 16+")
else:
    print(f"Grade {index}")

File: Week 2/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <math.h>

int main(void)
{

//promt user for text input
    string text = get_string("Enter Text Here: \n");

//count the number of letters
    int letters = 0;
    for (int i = 0; i < strlen(text); i++)
    {
        if ((text[i] >= 'a' && text[i] <= 'z') || (text[i] >= 'A' && text[i] <= 'Z'))
        {
            letters++;
        }

    }
    printf("\n");

    //count the number of words
    int words = 1;
    for (int i = 0; i < strlen(text); i++)
    {
        if (text[i] == ' ')
        {
            words++;
        }

    }

    //count the number of sentences
    int sentences = 0;
    for (int i = 0; i < strlen(text); i++)
    {
        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            sentences++;

        }
    }

    //grade level index
    float calculation = (0.0588 * letters / words * 100) - (0.296 * sentences / words * 100) - 15.8;
    int index = round(calculation);

    // if text is less than 1
    if (index < 1)
    {
        printf("Before Grade 1\n");
        return 0;
    }
    // if text is greater than 16
    else if (index >= 16)
    {
        printf("Grade 16+\n");
        return 0;
    }
    else
    {
        //print grade number based on index
        printf("Grade %i\n", index);
    }
}